File: src/BaekJoon/Java/baekjoon_1110.java
package BaekJoon.Java;
// https://www.acmicpc.net/problem/1110

import java.util.Scanner;

public class baekjoon_1110 {
	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		int n = sc.nextInt();
		int front = 0;
		int rear = 0;
		int count = 0;
		int answer = n;

		while(true) {
			front = n / 10;
			rear = n % 10;
			n = (rear * 10) + (front + rear) % 10;
			count++;

			if(answer == n)
				break;
		}

		System.out.println(count);
		sc.close();
	}
}

File: src/BaekJoon/Rust/baekjoon_1110.rs
use std::io::{BufRead, stdin};

pub fn baekjoon_1110() {
    let mut n: i32 = stdin().lock().lines().next().unwrap().unwrap().trim().parse::<i32>().unwrap();
    let mut count: i32 = 0;
    let answer: i32 = n;
    
    loop {
        let front: i32 = n / 10;
        let rear: i32 = n % 10;
        
        n = (rear * 10) + (front + rear) % 10;
        count += 1;
        
        if answer == n {
            break;
        }
    }
    
    println!("{}", count);
}